ui/src/app/utils/index.spec.tsx

```diff
@@ -0,0 +1,27 @@
+import * as React from 'react';
+
+import {apiCallWithGatewayTimeoutRetries} from 'app/utils/index';
+
+const functionStub = {
+  successfulFunction() {
+    return Promise.resolve();
+  },
+  failedFunction() {
+    return Promise.resolve({status: 504}).then(response => {throw response});
+  }
+}
+
+
+describe('IndexUtils', () => {
+  it('should not retry if successful', async() => {
+    const successfulFunctionSpy = spyOn(functionStub, 'successfulFunction').and.callThrough();
+    await apiCallWithGatewayTimeoutRetries(() => functionStub.successfulFunction());
+    expect(successfulFunctionSpy).toHaveBeenCalledTimes(1);
+  });
+
+  it('should retry three times by default', async() => {
```

Comment:
Nice test!

Super nitpick: maybe qualify the test description with "apiCallWithGatewayTimeout" since index.tsx has a bunch of other stuff which might eventually get tested too.